src/components/Carousel.vue

```diff
@@ -0,0 +1,276 @@
+        const isOldSelectionValid = this.entries.indexOf(oldSelection) !== -1;
+        if (isNewSelectionValid) {
+            this.effectiveSelected = newSelection;
+        } else if (!isOldSelectionValid) {
+            this.effectiveSelected = this.entries[0];
+        } // else keep the old selection
+
+        if (this.effectiveSelected !== oldSelection) {
+            this.$emit('select', this.effectiveSelected);
+        }
+    }
+
+    @Watch('entryMargin')
+    private async _updateDimensions(newWatcherValueOrTween = true) {
+        const tween = typeof newWatcherValueOrTween === 'boolean' ? newWatcherValueOrTween : true;
+        await Vue.nextTick(); // let Vue render new entries
+        let largestHeight = 0;
+        let largestMinDistance = 0;
+        for (let i = 0; i < this.entries.length; ++i) {
+            const [ el1 ] = this.$refs[this.entries[i]];
+            const [ el2 ] = this.$refs[this.entries[(i + 1) % this.entries.length]];
+            largestHeight = Math.max(largestHeight, el1.offsetHeight);
+            const minDistance = el1.offsetWidth / 2 + el2.offsetWidth / 2 + this.entryMargin;
+            largestMinDistance = Math.max(largestMinDistance, minDistance);
+        }
+        // Choose radius big enough such that two items can be rendered side by side without overlapping.
+        // Calculate on a right triangle formed by radius, half distance and perpendicular from center point
+        // to distance line.
+        const centerAngle = 2 * Math.PI / this._totalPositionCount / 2; // angle at circle center point
+        const radius = (largestMinDistance / 2) / Math.sin(centerAngle);
+        this.radius.tweenTo(radius, tween ? this.animationDuration : 0);
+        (this.$el as HTMLElement).style.minHeight = `${largestHeight}px`;
+        this._rerender();
+    }
+
+    @Watch('effectiveSelected')
+    @Watch('disabled')
+    private _updateRotations(newWatcherValueOrTween = true, previousWatcherValue?) {
+        const tween = typeof newWatcherValueOrTween === 'boolean' && typeof previousWatcherValue === 'undefined'
+            ? newWatcherValueOrTween // specified whether to tween
+            : true; // did not specify whether to tween or method was called as a watcher (default to true)
+        // clean up removed entries
+        for (const entry of this.rotations.keys()) {
+            if (this.entries.indexOf(entry) !== -1) continue;
+            this.rotations.delete(entry);
+        }
+        // update rotations
+        for (const entry of this.entries) {
+            const rotation = this.rotations.get(entry) || new Tweenable();
+            const tweenTime = tween ? this.animationDuration : 0;
+            rotation.tweenTo(this._calculateTargetRotation(entry, rotation.currentValue), tweenTime);
+            this.rotations.set(entry, rotation);
+        }
+        this._rerender();
+    }
+
+    private _calculateTargetRotation(entry, currentRotation): number { // rotation in radians
```

Comment:
VSCode tells me that most of the arguments to methods in this component need to be typed (no implicit `any`).